src/Sulu/Bundle/SecurityBundle/Controller/ResettingController.php

```diff
@@ -0,0 +1,309 @@
+        $user->setTokenExpiresAt(null);
+        $user->setTokenEmailsSent(null);
+        $em->persist($user);
+        $em->flush();
+    }
+
+    /**
+     * Sends the password-reset-token of a user to an email-adress
+     * @param $user
+     * @param $from
+     * @param $to
+     * @throws NoTokenFoundException
+     * @throws TokenEmailsLimitReachedException
+     */
+    private function sendTokenEmail(User $user, $from, $to) {
+        if ($user->getPasswordResetToken() === null) {
+            throw new NoTokenFoundException($user);
+        }
+        if ($user->getTokenEmailsSent() === self::MAX_NUMBER_EMAILS) {
+            throw new TokenEmailsLimitReachedException(self::MAX_NUMBER_EMAILS, $user);
+        }
+        $mailer = $this->get('mailer');
+        $translator = $this->get('translator');
+        $em = $this->getDoctrine()->getManager();
+        $message = $mailer->createMessage()
+            ->setSubject(
+                $translator->trans(self::EMAIL_SUBJECT_KEY, array(), 'backend')
+            )
+            ->setFrom($from)
+            ->setTo($to)
+            ->setBody(
+                $translator->trans(self::EMAIL_MSG_KEY, array(), 'backend') . PHP_EOL .
+                $this->generateUrl('sulu_admin.reset', array('token' => $user->getPasswordResetToken()), true)
+            );
+        $mailer->send($message);
+        $user->setTokenEmailsSent($user->getTokenEmailsSent() + 1);
+        $em->persist($user);
+        $em->flush();
+    }
+
+    /**
+     * Changes the password of a user
+     * @param User $user
+     * @param string $password
+     * @throws MissingPasswordException
+     */
+    private function changePassword(User $user, $password) {
+        if ($password === '') {
+            throw new MissingPasswordException();
+        }
+        $em = $this->getDoctrine()->getManager();
+        $user->setPassword($this->encodePassword($user, $password, $user->getSalt()));
+        $em->persist($user);
+        $em->flush();
+    }
+
+    /**
+     * Generates a new token for a new user
+     * @param User $user
+     * @throws TokenAlreadyRequestedException
+     */
+    private function generateTokenForUser(User $user) {
+        if ($user->getPasswordResetToken() !== null &&
+            (new \DateTime())->add(self::getResetInterval()) < $user->getTokenExpiresAt()->add(self::getRequestInterval())
+        ) { // if a token was already requested within the request interval time frame
+            throw new TokenAlreadyRequestedException(self::getRequestInterval());
+        }
+        $em = $this->getDoctrine()->getManager();
+
+        $user->setPasswordResetToken($this->getToken());
+        $expireDateTime = (new \DateTime())->add(self::getResetInterval());
+        $user->setTokenExpiresAt($expireDateTime);
+        $user->setTokenEmailsSent(0);
+
+        $em->persist($user);
+        $em->flush();
+    }
+
+    /**
+     * Returns a unique token
+     * @return string - token
```

Comment:
Please remove the `-`, that's not a standard. And maybe you can write `@return string A unique token`? Just one word is a bit too less